schema_based_on_diagram.sql

```diff
@@ -0,0 +1,47 @@
+CREATE DATABASE clinic;
+
+CREATE TABLE patients(
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100),
+  date_of_birth DATE
+);
+
+CREATE TABLE medical_histories(
+  id SERIAL PRIMARY KEY,
+  admitted_at TIMESTAMP,
+  patient_id INT REFERENCES patients(id),
+  status VARCHAR(50)
+);
+
+CREATE TABLE invoices(
+  id SERIAL PRIMARY KEY,
+  total_amount DECIMAL(3, 2),
+  generated_at TIMESTAMP,
+  paid_at TIMESTAMP,
+  medical_history_id INT REFERENCES medical_histories(id)
+);
+
+CREATE TABLE treatments(
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(50),
+  name VARCHAR(100)
+);
+
+CREATE TABLE treatment_histories(
+  treatment_id INT REFERENCES treatments(id),
+  medical_history_id INT REFERENCES medical_histories(id)
```

Comment:
- [ ] Please make sure to create indices for the following columns since they show only the `id`.

Reply:
@ShadyShawkat for which columns should we create indices? Could you please clarify that a bit?
Because, I think, we've already created indices for all the required columns.

Waiting for your reply.

Reply:
Hi @safar1212, The columns are shown in the code snippet in this inline comment. The `id` columns for the `treatment_histories` table.